src/Operation/BulkWrite.php

```diff
@@ -191,10 +191,19 @@ public function __construct(string $databaseName, string $collectionName, array
                         throw InvalidArgumentException::invalidType(sprintf('$operations[%d]["%s"][1]', $i, $type), $args[1], 'array or object');
                     }
 
+                    // Treat empty arrays as replacement documents for BC
+                    if ($args[1] === []) {
+                        $args[1] = (object) $args[1];
+                    }
+
                     if (is_first_key_operator($args[1])) {
                         throw new InvalidArgumentException(sprintf('First key in $operations[%d]["%s"][1] is an update operator', $i, $type));
                     }
 
+                    if (is_pipeline($args[1], true /* allowEmpty */)) {
```

Comment:
I don't need such comment with phpstorm. It may be replaced by named argument with PHP 8 named argument when the lib is upgraged.

<img width="448" alt="image" src="https://github.com/mongodb/mongo-php-library/assets/400034/ce75c835-023d-4afd-93ba-629dc20f6beb">

Reply:
Interesting. How does PhpStorm decide to show the second parameter name but not the first (`$args[1]`)? It looks like it shows the first parameter for `sprintf()` on the second line.

Reply:
The hint is not shown when a variable is passed, it's shown for static values only.